### ISSUE TITLE
shell script not executable warning

### ISSUE THREAD

**Issue by fusionneur** [github/super-linter#1409]
trying to lint script.sh. I get this warning in github action log:

2021-03-27 02:17:05 [WARN]   Warnings found in [bash-exec] linter!
2021-03-27 02:17:05 [WARN]   Error: File:[script.sh] is not executable

What should one do to fix this?

**Comment by nvuillam** [comment]
You can run `chmod +x script.sh` :) if you are on Windows, you may use [Git Bash](https://gitforwindows.org/) :)

**Comment by vrealizebuildtools** [comment]
Why it needs to be executable?

**Comment by fusionneur** [comment]
why it needs to be executable?

**Comment by nvuillam** [comment]
Because on a linux environment, you can't run a .sh if it is not executable :)

**Comment by vrealizebuildtools** [comment]
lawl i know that, but for checking syntax/linting why the script needs to run ?

**Comment by fusionneur** [comment]
lawl i know that, but for checking syntax/linting why the script needs to run ?

**Comment by nvuillam** [comment]
The .sh scripts are not run with SuperLinter, they are just checked to be runnable, because if they are not, when they are supposed to run later, it will crash ^^

**Comment by fusionneur** [comment]
In any case, my script already had x flag in permissions (using cygwin). See example in my repo:

https://github.com/fusionneur/ethernally/runs/2212703941?check_suite_focus=true


Repo seen from cygwin:

```
$ ll
total 51
drwxrwx---+ 1 Fusion None     0 Mar 26 20:28 .
drwxrwx---+ 1 Fusion None     0 Mar 25 19:37 ..
drwxrwx---+ 1 Fusion None     0 Mar 28 17:08 .git
-rwxrwx---+ 1 Fusion None   444 Nov  3 15:08 .gitattributes
drwxrwx---+ 1 Fusion None     0 Mar 26 18:28 .github
-rwxrwx---+ 1 Fusion None    40 Dec  9 12:01 .gitignore
-rwxrwx---+ 1 Fusion None  3428 Nov  3 12:26 CODE_OF_CONDUCT.md      
-rwxrwx---+ 1 Fusion None  1151 Nov  3 12:26 CONTRIBUTING.md
-rwxrwx---+ 1 Fusion None  1601 Mar 25 19:54 ethernally.lnk
-rwxrwx--x+ 1 Fusion None 14978 Mar 28 17:08 ethernally.sh
-rw-rw-r--+ 1 Fusion None   129 Mar 27 15:25 last_working_device.conf
-rwxrwx---+ 1 Fusion None  1094 Nov  3 12:26 LICENSE
-rwxrwx---+ 1 Fusion None  2545 Mar 27 04:35 README.md
```

**Comment by nvuillam** [comment]
That's indeed strange :/

Did `git update-index --chmod=+x script.sh` solve the problem ?

**Comment by fusionneur** [comment]
Yes, warning is gone using `git update-index --chmod=+x script.sh`
I had to make a slight modification in the script so that git discovers it because just using the above git command didn't trigger a change.

**Comment by zkoppert** [comment]
Looks like this one is solved! Thanks @nvuillam!

**Comment by nvuillam** [comment]
@zkoppert you're welcome :)
